src/Kentico.Xperience.Shopify/Products/ShopifyPriceService.cs

```diff
@@ -1,62 +1,69 @@
 
 internal class ShopifyPriceService : ShopifyServiceBase, IShopifyPriceService
 {
-    private readonly IProductService productService;
-    private readonly IShopifyIntegrationSettingsService settingsService;
+    private readonly IGraphService graphService;
 
     public ShopifyPriceService(
         IShopifyIntegrationSettingsService integrationSettingsService,
-        IProductServiceFactory productServiceFactory,
-        IShopifyIntegrationSettingsService settingsService)
+        IGraphServiceFactory graphServiceFactory)
         : base(integrationSettingsService)
     {
-        productService = productServiceFactory.Create(shopifyCredentials);
-        this.settingsService = settingsService;
+        graphService = graphServiceFactory.Create(shopifyCredentials);
     }
 
-    public async Task<IDictionary<string, ProductPriceModel>> GetProductsPrice(IEnumerable<string> shopifyProductIds)
+    public async Task<IDictionary<string, ProductPriceModel>> GetProductsPrice(IEnumerable<string> shopifyProductIds, CountryCode countryCode)
     {
         return await TryCatch(
-            async () => await GetProductsPriceInternal(shopifyProductIds),
+            async () => await GetProductsPriceInternal(shopifyProductIds, countryCode),
             () => new Dictionary<string, ProductPriceModel>());
     }
 
-    private async Task<IDictionary<string, ProductPriceModel>> GetProductsPriceInternal(IEnumerable<string> shopifyProductIds)
-    {
-        string currencyCode = settingsService.GetWebsiteChannelSettings()?.CurrencyCode ?? string.Empty;
-
-        var dict = new Dictionary<string, ProductPriceModel>();
-
-        var filter = new ProductListFilter()
-        {
-            Ids = shopifyProductIds.Select(long.Parse),
-            Fields = "Variants,Id",
-            PresentmentCurrencies = [currencyCode]
+    private async Task<IDictionary<string, ProductPriceModel>> GetProductsPriceInternal(IEnumerable<string> shopifyProductIds, CountryCode countryCode)
+    {
+        var productsQuery = shopifyProductIds.Select(x => $"id:{x}")
+            .Join(" OR ");
+
+        var request = new GraphRequest
+        {
+            Query = "query getProductPrices($query:String!,$country:CountryCode){products(first:250,query:$query){nodes{id hasOnlyDefaultVariant contextualPricing(context:{country:$country}){minVariantPricing{price{amount currencyCode}compareAtPrice{amount currencyCode}}}}}}",
+            Variables = new Dictionary<string, object>
+            {
+                ["query"] = productsQuery,
+                ["country"] = countryCode.ToStringRepresentation()
+            }
         };
-        var result = await productService.ListAsync(filter, true);
+
+        var result = await graphService.PostAsync<ProductConnectionResult>(request);
+        var products = result.Data.Products.nodes;
 
-        foreach (var product in result.Items)
+        var dict = new Dictionary<string, ProductPriceModel>();
+        if (products == null)

```

Comment:
It's recommended to use `is` operator for null-checks in c#. 

NOTE: Look at all nullchecks in this file.